src/sparkx/Oscar.py

```diff
@@ -242,6 +242,29 @@ def _particle_as_list(self, particle: Any) -> List[Union[float, int]]:
             )
 
         return particle_list
+    
+    def _update_after_merge(self, other: BaseStorer) -> None:
+        """
+        Updates the current instance after merging with another Oscar instance.
+
+        This method is called after merging two Oscar instances to update the
+        attributes of the current instance based on the attributes of the other instance.
+
+        Parameters
+        ----------
+        other : Oscar
+            The other Oscar instance that was merged with the current instance.
+
+        Raises
+        ------
+        UserWarning
+            If the Oscar formats of the two instances do not match, a warning is issued.
+        """
+        if not isinstance(other, Oscar):
+            raise TypeError("Can only add Oscar objects to Oscar.")
+        if self.oscar_format_ != other.oscar_format_:
+            warnings.warn("Oscar format of the merged instances do not match. Taking the left-hand side Oscar format.")
```

Comment:
Same as in Jetscape. Should this be an error?

Reply:
Plenty of the OscarFormats are compatible. I think there should be only an issue with IC I guess, as there the t/tau definition changes.

Reply:
Hmmmm...but if this is an issue depends on the intended analysis. Maybe I just want to get the v2. Then this would be fine.

Reply:
That's also true. So we leave this to the user to make sense of the loaded data and the keys needed for the analysis.